primitives/core/src/testing.rs

```diff
@@ -131,6 +135,62 @@ impl crate::traits::BareCryptoStore for KeyStore {
 	fn has_keys(&self, public_keys: &[(Vec<u8>, KeyTypeId)]) -> bool {
 		public_keys.iter().all(|(k, t)| self.keys.get(&t).and_then(|s| s.get(k)).is_some())
 	}
+
+	fn supported_keys(&self, id: KeyTypeId, keys: Vec<CryptoTypePublicPair>) -> Result<Vec<CryptoTypePublicPair>, String> {
+		let ed25519_existing_keys: Vec<Vec<u8>> = self.ed25519_public_keys(id).iter()
+								.map(|k| k.to_raw_vec()).collect();
+		let sr25519_existing_keys: Vec<Vec<u8>> = self.sr25519_public_keys(id).iter()
+								.map(|k| k.to_raw_vec()).collect();
+
+		Ok(keys.iter().filter_map(|k| {
+			match k.0 {
+				sr25519::SR25519_CRYPTO_ID if sr25519_existing_keys.contains(&k.1.to_vec()) => Some(k),
+				ed25519::ED25519_CRYPTO_ID if ed25519_existing_keys.contains(&k.1.to_vec()) => Some(k),
+				_ => None
+			}
+		}).cloned().collect::<Vec<_>>())
```

Comment:
```suggestion
		}).collect::<Vec<_>>())
```

Or better just use set intersection again.